internal/provider/resource_license_assignment.go

```diff
@@ -0,0 +1,213 @@
+package googleworkspace
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	licensing "google.golang.org/api/licensing/v1"
+)
+
+func resourceLicenseAssignment() *schema.Resource {
+	return &schema.Resource{
+		Description: "License Assignment Resource",
+		CreateContext: resourceLicenseAssignmentCreate,
+		ReadContext:   resourceLicenseAssignmentRead,
+		UpdateContext: resourceLicenseAssignmentUpdate,
+		DeleteContext: resourceLicenseAssignmentDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+		    "user_email": {
+            	Description: "The user's current primary email address.",
+            	Type:        schema.TypeString,
+            	Required:    true,
+            },
+            "sku_id": {
+                Description: "A product SKU's unique identifier.",
+                Type:        schema.TypeString,
+                Required:    true,
+            },
+            "product_id": {
+                 Description: "A product's unique identifier.",
+                 Type:        schema.TypeString,
+                 Required:    true,
+            },
+		},
+	}
+}
+
+func resourceLicenseAssignmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	client := meta.(*apiClient)
+
+	productId := d.Get("product_id").(string)
+	skuId := d.Get("sku_id").(string)
+	userId := d.Get("user_email").(string)
+
+	licensingService, diags := client.NewLicensingService()
+	if diags.HasError() {
+		return diags
+	}
+
+	licenseAssignmentsService, diags := GetLicenseAssignmentsService(licensingService)
+	if diags.HasError() {
+		return diags
+	}
+
+	la := &licensing.LicenseAssignmentInsert{
+		UserId: userId,
+	}
+
+	retryErr := resource.Retry(5*time.Minute, func() *resource.RetryError {
+	    _, err := licenseAssignmentsService.Insert(productId, skuId, la).Do()
+	    if err != nil {
+	        if strings.Contains(err.Error(), "Error 403") == true {
```

Comment:
Our retry transport should already be checking for retryable errors (https://github.com/hashicorp/terraform-provider-googleworkspace/blob/d7213bfc984bbde27c1afcb38351328ec9cc9b58/internal/provider/retry_predicates.go#L102).

If you want to look at an example of how we typically handle retries, you can look at [an example here](https://github.com/hashicorp/terraform-provider-googleworkspace/blob/5c3f1021ffafcc1c9dbc35bacb1a28624f1f4f03/internal/provider/resource_schema.go#L185).